data/custom-formats.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright 2024 Commaille Kévin
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# The list of custom formats supported for the `format` key and the
+# `x-pattern-format` extension (see `openapi_extensions.md` for more details).
+#
+# Each entry must use the `mx-` prefix and have the form:
+#
+# mx-custom-key:
+#   title: The title rendered in the specification
+#   url: /url/to#definition
+#   pattern: "REGEX" (not used for rendering, but allows to define the common
+#                     pattern to use for the format)
```

Comment:
I'm minded to agree with @KitsuneRal that this might be more of a hindrance than a help. Maybe we could just use yaml comments: `# Regex: "^@"`

Reply:
I put the regex in a comment.